python/ray/_raylet.pyx

```diff
@@ -480,32 +479,36 @@ cdef execute_task(
         const c_vector[CObjectID] &c_return_ids,
         c_vector[shared_ptr[CRayObject]] *returns):
 
-    worker = ray.worker.global_worker
-
-    actor_id = worker.core_worker.get_actor_id()
-    job_id = worker.core_worker.get_current_job_id()
-    task_id = worker.core_worker.get_current_task_id()
+    cdef object worker = ray.worker.global_worker
+    cdef object manager = worker.function_actor_manager
+    cdef CoreWorker core_worker = worker.core_worker
+    cdef JobID job_id = core_worker.get_current_job_id()
+    cdef CTaskID task_id = core_worker.core_worker.get().GetCurrentTaskId()
 
     # Automatically restrict the GPUs available to this task.
     ray.utils.set_cuda_visible_devices(ray.get_gpu_ids())
 
-    function_descriptor = FunctionDescriptor.from_bytes_list(
-        ray_function.GetFunctionDescriptor())
+    cdef const c_vector[c_string]* d = &ray_function.GetFunctionDescriptor()
```

Comment:
The ugliness in this paragraph is because Cython doesn't support references. But avoiding the vector copies does shave more than 10us from the execution code path, so might be worth it. Maybe there is a better way to do it.